pkg/virt-api/webhooks/defaults.go

```diff
@@ -168,6 +170,10 @@ func setDefaultResourceRequests(clusterConfig *virtconfig.ClusterConfig, spec *v
 			overcommit := clusterConfig.GetMemoryOvercommit()
 			if overcommit == 100 {
 				resources.Requests[k8sv1.ResourceMemory] = *memory
+			} else if overcommit > 100 && guestMemory && util.IsVFIOVMI(spec) {
+				// ignore memory overcommit if it results in guest memory > requests
+				// when VFIO device is requested
+				resources.Requests[k8sv1.ResourceMemory] = *memory
```

Comment:
We need at least a warning here, telling the user we ignored an explicit CR setting.